Source/Core/Core/IOS/USB/Emulated/LogitechMic.cpp

```diff
@@ -0,0 +1,707 @@
+    0x04, /* bDescriptorType */
+    0x01, /* bInterfaceNumber */
+    0x01, /* bAlternateSetting */
+    0x01, /* bNumEndpoints */
+    0x01, /* bInterfaceClass */
+    0x02, /* bInterfaceSubClass */
+    0x00, /* bInterfaceProtocol */
+    0x00, /* iInterface */
+
+    /* Audio Streaming Interface */
+    0x07,       /* bLength */
+    0x24,       /* bDescriptorType */
+    0x01,       /* bDescriptorSubtype */
+    0x0a,       /* bTerminalLink */
+    0x00,       /* bDelay */
+    0x01, 0x00, /* wFormatTag */
+
+    /* Audio Type I Format */
+    0x17,             /* bLength */
+    0x24,             /* bDescriptorType */
+    0x02,             /* bDescriptorSubtype */
+    0x01,             /* bFormatType */
+    0x01,             /* bNrChannels */
+    0x02,             /* bSubFrameSize */
+    0x10,             /* bBitResolution */
+    0x05,             /* bSamFreqType */
+    0x40, 0x1f, 0x00, /* tSamFreq 1 */
+    0x11, 0x2b, 0x00, /* tSamFreq 2 */
+    0x22, 0x56, 0x00, /* tSamFreq 3 */
+    0x44, 0xac, 0x00, /* tSamFreq 4 */
+    0x80, 0xbb, 0x00, /* tSamFreq 5 */
+
+    /* Endpoint - Standard Descriptor */
+    0x09,       /* bLength */
+    0x05,       /* bDescriptorType */
+    0x84,       /* bEndpointAddress */
+    0x0d,       /* bmAttributes */
+    0x60, 0x00, /* wMaxPacketSize */
+    0x01,       /* bInterval */
+    0x00,       /* bRefresh */
+    0x00,       /* bSynchAddress */
+
+    /* Endpoint - Audio Streaming */
+    0x07,      /* bLength */
+    0x25,      /* bDescriptorType */
+    0x01,      /* bDescriptor */
+    0x01,      /* bmAttributes */
+    0x02,      /* bLockDelayUnits */
+    0x01, 0x00 /* wLockDelay */
+};
+
+static constexpr u16 LogitechUSBHDR(u8 dir, u8 type, u8 recipient, RequestCode request)
+{
+  return USBHDR(dir, type, recipient, Common::ToUnderlying(request));
+}
+
+int LogitechMic::SubmitTransfer(std::unique_ptr<CtrlMessage> cmd)
+{
+  // Reference: https://www.usb.org/sites/default/files/audio10.pdf
+  DEBUG_LOG_FMT(IOS_USB,
+                "[{:04x}:{:04x} {}:{}] Control: bRequestType={:02x} bRequest={:02x} wValue={:04x}"
+                " wIndex={:04x} wLength={:04x}",
+                m_vid, m_pid, m_index, m_active_interface, cmd->request_type, cmd->request,
+                cmd->value, cmd->index, cmd->length);
+  switch (cmd->request_type << 8 | cmd->request)
+  {
+  case USBHDR(DIR_DEVICE2HOST, TYPE_STANDARD, REC_DEVICE, REQUEST_GET_DESCRIPTOR):
+  {
+    // It seems every game always pokes this at first twice; one with a length of 9 which gets the
+    // config, and then another with the length provided by the config.
+    DEBUG_LOG_FMT(IOS_USB, "[{:04x}:{:04x} {}:{}] REQUEST_GET_DESCRIPTOR index={:04x} value={:04x}",
+                  m_vid, m_pid, m_index, m_active_interface, cmd->index, cmd->value);
+    cmd->FillBuffer(FULL_DESCRIPTOR.data(), std::min<size_t>(cmd->length, FULL_DESCRIPTOR.size()));
+    cmd->GetEmulationKernel().EnqueueIPCReply(cmd->ios_request, IPC_SUCCESS);
```

Comment:
I mentioned in discord, but I'll say it here: `EnqueueIPCReply` shouldn't be used in these functions. You should just return `IPC_SUCCESS` or whatever value you need and `EnqueueIPCReply` will happen automatically from the caller.

Reply:
The other USB emulated devices are doing something similar. However, they were doing so using the `ScheduleTransfer`/`ScheduleTransferCompletion` wrapper instead, while the WiiSpeak code doesn't _(regarding `ScheduleTransfer`)_. IIRC, it's because of the USB transfer logic in `USBHost::HandleTransfer`.